### PR TITLE
Implementation of method ping()

### PR DESCRIPTION
**Pull request by pkleanthous-zz**
This PR aim to provide the implementation of ping() method on Junos Devices.

It follows the guidelines for the method that you can find here: [napalm.readthedocs.io](http://napalm.readthedocs.io/en/latest/base.html)

More info about `ping` command at [Juniper Documentation](https://www.juniper.net/documentation/en_US/junos12.1x47/topics/task/operational/security-ping-command-using.html)

### CODE REVIEW

**Comment by dbarrosop** [comment]
Great job @pkleanthous! I think everything is fine but I will leave the approval to @mirceaulinic as he has more experience with this sort of operations in JunOS devices.

File: napalm_junos/junos.py [hunk 1]
```diff
@@ -1246,6 +1246,77 @@ def traceroute(self,
...
+        ping_dict = {}
+
+        source_str = ''
+        maxttl_str = ''
+        timeout_str = ''
+        size_str = ''
+        count_str = ''
+
+        if source:
+            source_str = 'interface {source}'.format(source=source)
```
Comment (mirceaulinic) [inline napalm_junos/junos.py]: Here should be `source_str = 'source {source}'.format(source=source)` instead - the `source` arg is accepted.

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
Thanks @pkleanthous!

It looks good on the surface - but I will make some tests later next week. Meanwhile, please apply the following changes:

- `source` arg, commented also inline
- wherever you have <object>.findtext replace it with the napalm-base helper `find_txt`: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/helpers.py#L142
Example:  `probe_summary.findtext("rtt-minimum")` would be replaced with `napalm_base.helpers.find_txt(probe_summary, 'rtt-minimum')` - the advantage being that will give you empty string instead of `None` in case the tag does not exist, which would lead to unwanted exceptions
- Similarly, please use the `convert` helper: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/helpers.py#L166 when casting data. Example: `int(<object>)`  --> `napalm_base.helpers.convert(int, <object>, 100)` which will return you 100 if <object> cannot be converted / is None etc.

**Comment by pkleanthous-zz** [comment]
Hi @dbarrosop & @mirceaulinic. Thanks for the review.

Just made the requested changes.

**Review by mirceaulinic — APPROVED** [review]

**Comment by XioNoX** [comment]
fwiw, I tried this pull request with the new validation class/framework (through napalm_validate).
Everything worked well!

With a valid destination IP:
```
 'ping': {u'complies': True,
          u'extra': [],
          u'missing': [],
          u'present': {'success': {u'complies': True, u'nested': True}}}}
```
With an invalid destination IP:
```
 'ping': {u'complies': False,
          u'extra': [],
          u'missing': ['success'],
          u'present': {}}}
```

**Pull request merged**